server/app/repository/ApiBridgeConfigurationRepository.java

```
@@ -0,0 +1,95 @@
+  }
+
+  /** Returns all {@link ApiBridgeConfigurationModel} records asynchronously. */
+  public CompletionStage<ImmutableList<ApiBridgeConfigurationModel>> findAll() {
+    return supplyAsync(
+        () ->
+            database
+                .find(ApiBridgeConfigurationModel.class)
+                .setLabel("ApiBridgeConfigurationRepository.findAll")
+                .setProfileLocation(queryProfileLocationBuilder.create("findAll"))
+                .findList()
+                .stream()
+                .collect(ImmutableList.toImmutableList()),
+        dbExecutionContext);
+  }
+
+  /** Returns an {@link ApiBridgeConfigurationModel} record or empty optional asynchronously. */
+  public CompletionStage<Optional<ApiBridgeConfigurationModel>>
+      findByHostUrlAndUrlPathAndCompatibilityLevel(
+          String hostUrl, String urlPath, CompatibilityLevel compatibilityLevel) {
+    return supplyAsync(
+        () ->
+            database
+                .find(ApiBridgeConfigurationModel.class)
+                .where()
+                .eq("host_url", hostUrl)
+                .and()
+                .eq("url_path", urlPath)
+                .and()
+                .eq("compatibilityLevel", compatibilityLevel)
+                .setLabel(
+                    "ApiBridgeConfigurationRepository.findByHostUrlAndUrlPathAndCompatibilityLevel")
+                .setProfileLocation(
+                    queryProfileLocationBuilder.create(
+                        "findByHostUrlAndUrlPathAndCompatibilityLevel"))
+                .findOneOrEmpty(),
+        dbExecutionContext);
+  }
+
+  /** Insert a new {@link ApiBridgeConfigurationModel} record asynchronously. */
+  public CompletionStage<ApiBridgeConfigurationModel> insert(ApiBridgeConfigurationModel model) {
+    return supplyAsync(
+        () -> {
+          database.insert(model);
+          return model;
+        },
+        dbExecutionContext);
+  }
+
```

`database.update(model)` may silently fail if the entity doesn't exist in the DB. Consider checking for existence before updating.

---

That will be handled at a higher layer